core/ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/dropdown/DropDownWidgetImpl.java

```diff
@@ -0,0 +1,289 @@
+
+
+    @AssistedInject
+    public DropDownWidgetImpl(Resources resources, ActionManager actionManager, KeyBindingAgent keyBindingAgent,
+                              Provider<PerspectiveManager> managerProvider, @NotNull @Assisted String listId) {
+        this.resources = resources;
+        this.listId = listId;
+
+        initWidget(UI_BINDER.createAndBindUi(this));
+
+        resources.dropdownListCss().ensureInjected();
+
+        listHeader.setStyleName(resources.dropdownListCss().menuElement());
+
+        marker.getElement().appendChild(resources.expansionImage().getSvg().getElement());
+        marker.addStyleName(resources.dropdownListCss().expandedImage());
+
+        addDomHandler(this, ClickEvent.getType());
+
+        this.actionManager = actionManager;
+        this.keyBindingAgent = keyBindingAgent;
+        this.managerProvider = managerProvider;
+
+        presentationFactory = new PresentationFactory();
+        actions = new DefaultActionGroup(actionManager);
+
+        Window.addResizeHandler(new ResizeHandler() {
+            @Override
+            public void onResize(ResizeEvent event) {
+                updatePopup();
+            }
+        });
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void selectElement(String id, String name) {
+        selectedId = id;
+        selectedName = name;
+        selectedElementName.setText(name == null ? "---" : name);
+    }
+
+    /** {@inheritDoc} */
+    @Nullable
+    @Override
+    public String getSelectedName() {
+        return selectedName;
+    }
+
+    public String getSelectedId() {
+        return selectedId;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void onClick(ClickEvent event) {
+        int left = getAbsoluteLeft();
+        int top = getAbsoluteTop() + listHeader.getOffsetHeight();
+        int width = listHeader.getOffsetWidth();
+        this.show(left, top, width, listId);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void updatePopup() {
+        if (popupMenu == null || !popupMenu.isAttached()) {
+            return;
+        }
+        this.hide();
+        int top = getAbsoluteTop() + listHeader.getOffsetHeight();
+        this.show(getAbsoluteLeft(), top, listHeader.getOffsetWidth(), listId);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Action createAction(String id, String name) {
+        return new SimpleListElementAction(id, name);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void onActionSelected(Action action) {
+        this.hide();
+    }
+
+    /**
+     * Shows a content menu and moves it to specified position.
+     *
+     * @param left
+     *         horizontal position
+     * @param top
+     *         vertical position
+     * @param width
+     *         header width
+     * @param itemId
+     *         list identifier
+     */
+    private void show(final int left, final int top, final int width, @NotNull String itemId) {
```

Comment:
Cleanup arguments, you don't have anonymous classes inside this method, so you don't need to mark them as final.